src/goto-programs/goto_trace.cpp

```diff
@@ -295,91 +377,107 @@ void show_goto_trace(
       break;
 
     case goto_trace_stept::typet::ASSIGNMENT:
-      if(step.pc->is_assign() ||
-         step.pc->is_return() || // returns have a lhs!
-         step.pc->is_function_call() ||
-         (step.pc->is_other() && step.lhs_object.is_not_nil()))
+      if(config.trace_config.show_value_assignments)
       {
-        if(prev_step_nr!=step.step_nr || first_step)
+        if(step.pc->is_assign() || step.pc->is_return()
```

Comment:
Suggest keeping old style for this, it's clearer when there's one disjunct per line